videostore/videostore.go

```diff
@@ -296,13 +296,20 @@ func (vs *videostore) Fetch(_ context.Context, r *FetchRequest) (*FetchResponse,
 		return nil, err
 	}
 	vs.logger.Debug("fetch command received")
-
 	fetchFilePath := generateOutputFilePath(
 		vs.config.Storage.OutputFileNamePrefix,
 		formatDateTimeToString(r.From),
 		"",
 		tempPath)
 
+	// Always attempt to remove the concat file after the operation.
+	// This handles error cases in Concat where it fails in the middle
+	// of writing.
+	defer func() {
```

Comment:
set up this defer after `Concat` succeeds, otherwise if `Concat` fails we won't produce a file & this will log an unnecessary warning.

Reply:
ah I see, never mind